### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by Raxerz**
This PR contains the following changes:
1. Material theme changes
2. Email template changes
3. Code refactoring

### CODE REVIEW

File: public/pages/forgot-password.html [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                                  </table>
+                                </td>
+                              </tr>
+                            </table>
+                          </td>
+                        </tr>
+                      </table>
+                      <p style="margin-top: 0; color: #74787E; font-size: 16px; line-height: 1.5em; text-align: left; font-family: Arial, 'Helvetica Neue', Helvetica, sans-serif; box-sizing: border-box;">If you did not request a password reset, please ignore this email or <a href="{{support_url}}" style="color: #3869D4; text-decoration: none; font-family: Arial, 'Helvetica Neue', Helvetica, sans-serif; box-sizing: border-box;">contact support</a> if you have questions.</p>
+                      <p style="margin-top: 0; color: #74787E; font-size: 16px; line-height: 1.5em; text-align: left; font-family: Arial, 'Helvetica Neue', Helvetica, sans-serif; box-sizing: border-box;">Thanks,
+                        <br>The Peace Corps Team</p>
```
Comment (Buddhiprabha) [inline public/pages/forgot-password.html]: Hard coded?

Comment (Raxerz) [inline public/pages/forgot-password.html]: Yes, should I change this?

Comment (visitha) [inline public/pages/forgot-password.html]: All HTML and JavaScript hard-coded strings should be in the property file.

Comment (Raxerz) [inline public/pages/forgot-password.html]: Oh okay @visitha. I have done it for angular files. Will do it for the node files as well.

File: public/pages/forgot-password.html [hunk 2]
```diff
@@ -0,0 +1,103 @@
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
```
Comment (Raxerz) [inline public/pages/forgot-password.html]: The css stylings for email templates have to be inlined for it to work, moving the css to a new file does not work well.

File: routes/api.js [hunk 3]
```diff
@@ -79,14 +84,15 @@ router.get('/username', authenticationHelpers.isAuthOrRedirect, (req, res) => {
     localUser.findAll({
         where: {
             email: email
```
Comment (Raxerz) [inline routes/api.js]: Spacing changes.

File: routes/api.js [hunk 4]
```diff
@@ -79,14 +84,15 @@ router.get('/username', authenticationHelpers.isAuthOrRedirect, (req, res) => {
...
-            if(error){
-                res.status(500).json({error: 'Something went wrong'});
-            }
-        });
+        }
+    }).then(data => {
+        const username = `${data[0].fname} ${data[0].lname}`;
+        res.status(200).json({username: username});
+    }).catch(error => {
+        if(error){
```
Comment (yukie1412) [inline routes/api.js]: spacing

File: routes/api.js [hunk 5]
```diff
@@ -162,7 +168,7 @@ router.patch('/updateProgressStatus', authenticationHelpers.isAuthOrRedirect, (r
                 const progressActivity = data.progress.activity;
                 const stageDiff = currStage - progressStage;
                 const activityDiff = currActivity - progressActivity;
-                if((currStage === progressStage || stageDiff === 1) && (activityDiff === 1)) {
+                if((currStage === progressStage && activityDiff === 1) || (stageDiff === 1 && (progressActivity === 3 || progressActivity === 0))) {
```
Comment (yukie1412) [inline routes/api.js]: spacing

File: routes/api.js [hunk 6]
```diff
@@ -172,48 +178,48 @@ router.patch('/updateProgressStatus', authenticationHelpers.isAuthOrRedirect, (r
                         }
                     })
                         .then(response => {
-                            return res.status(200).json({info: 'success'});
+                            return res.status(200).json({info: successCode.PCS003.message, code: successCode.PCS003.code});
                         })
-                } else if(stageDiff > 1 && activityDiff > 1) {
-                    return res.status(200).json({info: 'Illegal operation'});
                 } else if(stageDiff < 1 && activityDiff < 1) {
```
Comment (yukie1412) [inline routes/api.js]: spacing

File: routes/api.js [hunk 7]
```diff
@@ -172,48 +178,48 @@ router.patch('/updateProgressStatus', authenticationHelpers.isAuthOrRedirect, (r
...
 router.get('/infokitactive', authenticationHelpers.isAuthOrRedirect, (req, res) => {
     if(!req.user.email) {
-        return res.status(400).json({error: 'Email not provided'});
+        return res.status(400).json({error: errorCode.PCE021.message, code: errorCode.PCE021.code});
     }
     localUser.find({where: {
         email: req.user.email
     }}, {raw: true})
         .then(data => {
             if(!data) {
```
Comment (yukie1412) [inline routes/api.js]: spacing (there are a lot more in this file)

File: routes/utilityfunctions.js [hunk 8]
```diff
@@ -19,9 +21,23 @@ function validatePassword(password) {
     return re.test(password);
 }
 
+function readHTMLFile(path, callback) {
+    fs.readFile(path, {encoding: 'utf-8'}, function(err, html) {
+        if (err) {
+            throw err;
+            callback(err);
```
Comment (yukie1412) [inline routes/utilityfunctions.js]: have you tested if this works?

File: src/app/app.component.html [hunk 9]
```diff
@@ -1,15 +1,10 @@
...
-    <router-outlet></router-outlet>
-  </div>
+<!--The whole content below can be removed with the new code.-->
+<div class="header" [hidden]='!loggedIn'>
+    <app-header (toggleNavbar)="nav.toggle()"> </app-header>
+</div>
+<app-infokit #info > </app-infokit>
+<app-navbar #nav (togglePosition)="toggle();" (infoPop)="info.pop()" [hidden]='!loggedIn'> </app-navbar>
+<router-outlet></router-outlet>
+<div class="footer" [hidden]='!loggedIn || isMenuPage' style="z-index: 999">
```
Comment (yukie1412) [inline src/app/app.component.html]: inline css

File: src/app/header/header.component.html [hunk 10]
```diff
@@ -1,13 +1,11 @@
-<div class="container-fluid">
-  <div class="page-header header-bar row">
-    <div class="col-xs-2 col-sm-5">
-      <button type="button" class="btn btn-default btn-lg toggle-button" aria-label="Left Align" (click)="navbartoggle()">
+<md-toolbar color="primary">
+    <div class="col-xs-1">
+        <button type="button" class="btn btn-default btn-lg toggle-button" aria-label="Left Align" (click)="navbartoggle()">
         <span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span>
```
Comment (yukie1412) [inline src/app/header/header.component.html]: spacing

File: src/app/introduction/introduction.component.html [hunk 11]
```diff
@@ -1,8 +1,6 @@
-<div class="container-fluid">
-  <div class="row">
-    <div class="text-center">
-      <h1 class="stage-header">{{language?.heading}}</h1>
+<div class="container">
+    <div class="row">
+		<div class="col-md-12 text-center"><h2 class="stage-header">{{language?.heading}}</h2></div>
```
Comment (yukie1412) [inline src/app/introduction/introduction.component.html]: spacing

File: src/app/introduction/introduction.component.html [hunk 12]
```diff
@@ -1,8 +1,6 @@
...
     </div>
-  </div>
-  <router-outlet></router-outlet>
+  	<router-outlet></router-outlet>
```
Comment (yukie1412) [inline src/app/introduction/introduction.component.html]: spacing

File: src/app/malaria-101/malaria-101.component.html [hunk 13]
```diff
@@ -1,8 +1,7 @@
-<div class="container-fluid">
-    <div class="row">
-        <div class="text-center">
-            <h1 class="stage-header">{{language?.heading}}</h1>
-        </div>
-    </div>
-    <router-outlet></router-outlet>
+
+<div class="row">
+	<div class="col-md-12 text-center"><h2 class="stage-header">{{language?.heading}}</h2></div>
```
Comment (yukie1412) [inline src/app/malaria-101/malaria-101.component.html]: spacing

File: src/app/navbar/navbar.component.html [hunk 14]
```diff
@@ -1,19 +1,20 @@
 <div class="container-fluid">
   <div class="sidebar" [@toggleNav]="state">
-    <div class="sidebar-header">
-      <button type="button" class="btn btn-default btn-lg toggle-button" aria-label="Left Align" (click)="toggle()">
-        <span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span>
+      <md-toolbar color="primary">
+            <button type="button" class="btn btn-default btn-lg toggle-button" aria-label="Left Align" (click)="toggle()">
+      <span class="glyphicon glyphicon-align-justify" aria-hidden="true"></span>
```
Comment (yukie1412) [inline src/app/navbar/navbar.component.html]: spacing

File: src/app/navbar/navbar.component.scss [hunk 15]
```diff
@@ -16,3 +16,6 @@
     }
 }
 
+li {
+	padding: 10px;
```
Comment (yukie1412) [inline src/app/navbar/navbar.component.scss]: spacing

File: src/app/styles/modules/_colors.scss [hunk 16]
```diff
@@ -13,3 +13,16 @@ $dark-grey: #ECECEC;
...
 $dark-orange: #EBBB67;
+$yellow: #FBF38F;
+$orange: #FAE3BB;
+$dark-orange: #EBBB67;
+
+$intro-color: #00F977;
+$malaria-color: #EB5A61;
+$med-color: #017E84;
+$pc-policy-color: #F2AE3E;
+$white-color: #FFFFFF;
```
Comment (yukie1412) [inline src/app/styles/modules/_colors.scss]: how is this different from line 2?

File: routes/registration.js [hunk 17]
```diff
@@ -14,42 +14,76 @@ const validateEmail = utilityFunctions.validateEmail;
...
-    mail.mailOptions.html = `<b>Click on the link to complete the verification</b> <a href='${config.basePath}verification?token=${rString}&user=${req.body.email}'>Verify</a>`;
-    mail.smtpTransport.sendMail(mail.mailOptions, function(error) {
-        if (error) {
-            res.status(500).json({error: 'Something Went Wrong! Try again later.'});
-        } else {
-            res.json('Verification Mail Sent, Please check your mail.');
-        }
+    readHTMLFile('./public/pages/registration-verification.html', function(err, html) {
+        const template = handlebars.compile(html);
+        const replacements = {
```
Comment (yukie1412) [inline routes/registration.js]: this looks awfully similar among various places. ever thought of generalizing it in a function?

File: src/app/button-nav/button-nav.component.html [hunk 18]
```diff
@@ -0,0 +1,15 @@
+<span [className]="position + ' nav-back'" *ngIf="prevurl">
+    <button md-fab class="prev" (click)="prev()">
+        <i class="material-icons">arrow_back</i>
+    </button>
+</span>    
+<span class="nav-forward" *ngIf="completed && nexturl">
+    <button md-fab class="next"  (click)="next()">
```
Comment (yukie1412) [inline src/app/button-nav/button-nav.component.html]: spacing

File: src/app/button-nav/button-nav.component.html [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+        <i class="material-icons">arrow_forward</i>
+    </button>
+</span>
+<span class="nav-forward bounce" *ngIf="!completed && complete && nexturl">
+    <button md-fab class="next"  (click)="next()">
```
Comment (yukie1412) [inline src/app/button-nav/button-nav.component.html]: spacing

File: src/app/introduction/activity-1/activity-1.component.html [hunk 20]
```diff
@@ -0,0 +1,22 @@
...
+    <div class="col-lg-5 col-md-offset-4">
+        <blockquote class="rectangle-speech-border">
+            <p class="lead-title">
+                {{language?.descriptions.bodyContent1}}
+            </p>
+            <p class="lead-sub-heading">
+                {{language?.descriptions.bodyContent2}}
+            </p>
+            <p class="lead">
+              {{language?.descriptions.bodyContent3}}
```
Comment (yukie1412) [inline src/app/introduction/activity-1/activity-1.component.html]: spacing

File: src/app/malaria-101/activity-1/activity-1-1.component.html [hunk 21]
```diff
@@ -1,6 +1,11 @@
 <div class="row">
+    <div class="col-lg-6 col-lg-offset-3">
+        <h2 class="sub-heading-1 text-center">{{language?.heading}}</h2>
+    </div>
+</div>
+<div class="row">
 	<div class="col-md-4 col-md-offset-3">
```
Comment (yukie1412) [inline src/app/malaria-101/activity-1/activity-1-1.component.html]: tab?

File: src/app/meds-n-labels/activity-2/activity-2.component.ts [hunk 22]
```diff
@@ -138,19 +150,20 @@ export class MemoryGameComponent implements OnInit {
...
             }
             this._matches++;
             if (this._matches === 8) {
-                this._sharedData.customAlert('Good job!', 'You completed this activity!', 'success');
+                this._sharedData.customSuccessAlert(this.alerts.activitySuccessMsg, this.alerts.activitySuccessTitle);
                 this._status = {stage: 3, activity: 2};
                 this._dashboardService.updateProgressStatus(this._status).subscribe(response => {});
                 this.activityComplete = true;
             }
+            return;
```
Comment (yukie1412) [inline src/app/meds-n-labels/activity-2/activity-2.component.ts]: is this necessary?

File: src/index.html [hunk 23]
```diff
@@ -9,7 +9,7 @@
   <link rel="icon" type="image/x-icon" href="favicon.ico">
   <link rel="stylesheet" href="//netdna.bootstrapcdn.com/bootstrap/3.0.2/css/bootstrap.min.css"> <!-- load bootstrap css -->
   <link rel="stylesheet" href="//netdna.bootstrapcdn.com/font-awesome/4.0.3/css/font-awesome.min.css"> <!-- load fontawesome -->  
-  
+  <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons"> <!-- load fontawesome -->
```
Comment (yukie1412) [inline src/index.html]: mismatched comment

File: routes/registration.js (hunk 17 above)
Comment (Raxerz) [inline routes/registration.js]: I tried generalising it, but there are too many parameters to pass to the generalised function and they are different for different cases, so I decided not to go ahead with the generalising.

File: src/app/meds-n-labels/activity-2/activity-2.component.ts (hunk 22 above)
Comment (Raxerz) [inline src/app/meds-n-labels/activity-2/activity-2.component.ts]: Yes, I had removed it in my previous PR, but then I realised that it doesn't work well without it, it is to break out of the function execution when a certain condition satisfies.

File: src/app/authentication/login.component.ts [hunk 24]
```diff
@@ -12,26 +14,45 @@ import { AuthService } from '../services/auth.service';
...
+        });
         this.pcprepkitlogo = '../../assets/img/prepkitlogo.png';
     }
 
+    ngOnInit() {
+        const message = this._activatedRoute.snapshot.queryParams['msg'];
+        const error = this._activatedRoute.snapshot.queryParams['err'];
+        if (message) {
+            this._sharedData.customSuccessAlert(decodeURIComponent(message), 'Complete!');
+        } else if (error) {
```
Comment (yukie1412) [inline src/app/authentication/login.component.ts]: believe for convention, error is always handled first

File: src/app/services/shared.data.service.ts [hunk 25]
```diff
@@ -23,21 +26,20 @@ export class SharedDataService {
...
     }
 
     /**
      * Check progress of user (If the user has already completed the activity or not)
      */
-    checkProgress(currStage, currActivity): any {
-        let activityCompleted = false;
-        this._dashboardService.getProgressStatus().subscribe(response => {
+    checkProgress(currStage, currActivity, response): boolean {
             const activity = response.activity;
```
Comment (yukie1412) [inline src/app/services/shared.data.service.ts]: spacing

**Comment by Buddhiprabha** [comment]
I saw fullstops in only some of the error messages. Please make it consistent.

**Comment by Raxerz** [comment]
Resolved @yukie1412

**Comment by yukie1412** [comment]
conflicts

File: src/app/navbar/navbar.component.scss [hunk 26]
```diff
@@ -16,3 +16,6 @@
     }
 }
 
+li {
+    padding: 10px;
```
Comment (yukie1412) [inline src/app/navbar/navbar.component.scss]: 10px?

File: src/app/styles/partials/_figures.scss [hunk 27]
```diff
@@ -0,0 +1,60 @@
+.nav-forward {
+    position: fixed;
+    bottom: 40px;
+    right: 20px;
+    color: white;
```
Comment (yukie1412) [inline src/app/styles/partials/_figures.scss]: variable

File: src/app/styles/partials/_figures.scss [hunk 28]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+.nav-back {
+    position: fixed;
+    bottom: 40px;
+    left: 10px;
+}
+
+.prev {
```
Comment (yukie1412) [inline src/app/styles/partials/_figures.scss]: `.prev, .next`

**Comment by Raxerz** [comment]
Made the changes @yukie1412

File: .eslintignore [hunk 29]
```diff
@@ -0,0 +1,5 @@
+/dist
+
+# Files
+karma.conf.js
+protractor.conf.js
```
Comment (yukie1412) [inline .eslintignore]: newline

File: .eslintrc [hunk 30]
```diff
@@ -31,7 +31,6 @@
         "prefer-template": "error",
         "comma-spacing": ["error", { "before": false, "after": true }],
         "space-before-function-paren": ["error", "never"],
-        "space-before-blocks": ["error"],
```
Comment (yukie1412) [inline .eslintrc]: why removed?

Comment (Raxerz) [reply .eslintrc]: It was duplicated @yukie1412 . Please check line 30.

File: .eslintignore (hunk 29 above)
Comment (Raxerz) [reply .eslintignore]: Added new line

**Pull request merged**